pubsub/mqtt/mqtt.go

```diff
@@ -212,34 +212,23 @@ func (m *mqttPubSub) startSubscription(ctx context.Context) error {
 		// for backwards-compatibility; see: https://github.com/dapr/components-contrib/pull/2104
 		consumerClientID += "-consumer"
 	}
-	connCtx, connCancel := context.WithTimeout(ctx, defaultWait)
-	c, err := m.connect(connCtx, consumerClientID)
-	connCancel()
+
+	readyCh := make(chan struct{})
```

Comment:
should this be buffered ?
```suggestion
	readyCh := make(chan struct{}, 1)
```

Reply:
@ItalyPaleAle ^

Reply:
No, it's ok for this to be unbuffered so it blocks until it's consumed

Reply:
Changed to use `close()` rather than sending a message, which won't block